src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -262,16 +264,21 @@ public function setMethods($methods)
      */
     public function getResources()
     {
-        return array_unique($this->resources);
+        return array_values($this->resources);
     }
 
     /**
-     * Adds a resource for this collection.
+     * Adds a resource for this collection. If the resource already exists
+     * it is not added.
      *
      * @param ResourceInterface $resource A resource instance
      */
     public function addResource(ResourceInterface $resource)
     {
-        $this->resources[] = $resource;
+        $key = (string) $resource;
+
+        if (!array_key_exists($key, $this->resources)) {
```

Comment:
Maybe we should profile the proposed changes again now that we've complicated the logic to add these  `array_key_exists()` calls. Thanks!

Reply:
No need, just use isset. We know values in that array can not be null as we require object here.

Reply:
Check!